packages/components/src/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -0,0 +1,394 @@
+        return includeDataClasses ?
+            `${prefix}: ${name} (${containerPath})`:
+            name;
+    };
+
+    initParentOptions = (model: SampleTypeModel, responses: any[]) => {
+        let sets = List<IParentOption>();
+        responses.forEach((results) => {
+            const domain = fromJS(results.models[results.key]);
+
+            const prefix = results.key === DATA_CLASS_SCHEMA_KEY ? DATA_CLASS_IMPORT_PREFIX : SAMPLE_SET_IMPORT_PREFIX;
+            const labelPrefix = results.key === DATA_CLASS_SCHEMA_KEY ? "Data Class" : "Sample Set";
+
+            domain.forEach(row => {
+                const name = row.getIn(['Name', 'value']);
+                const containerPath = row.getIn(['Folder', 'displayValue']);
+                let label = NEW_SAMPLE_SET_OPTION && name === model.name ? NEW_SAMPLE_SET_OPTION.label : this.formatLabel(name, labelPrefix, containerPath);
+                sets = sets.push({
+                    value: prefix + name,
+                    label: label,
+                    schema: SCHEMAS.SAMPLE_SETS.SCHEMA,
+                    query: name, // Issue 33653: query name is case-sensitive for some data inputs (sample parents)
+                });
+            });
+        });
+
+        if(model.isNew()) {
+            sets = sets.push(NEW_SAMPLE_SET_OPTION);
+        }
+
+        this.mapParentAliasOptions(model, sets.sortBy(p => p.label, naturalSort) as List<IParentOption>);
+
+        //
+        //
+        // initSampleSetSelects(!model.isNew(), model.name, true).then(results => {
+        //     this.mapParentAliasOptions(results);
+        // }).catch((error) => {
+        //     this.onFinishFailure(resolveErrorMessage(error));
+        // });
+    };
+
+    mapParentAliasOptions = (model: SampleTypeModel, results: List<IParentOption>): void => {
+        const options = results.toArray();
+
+        let parentAliases = Map<string, IParentAlias>();
+
+        if (model && model.importAliases)
+        {
+            let initialAlias = Map<string,string>(model.importAliases);
+            initialAlias.forEach((val, key) => {
+                const newId = SampleTypeDesigner.generateAliasId();
+                parentAliases = parentAliases.set(newId, {
+                    id: newId,
+                    alias: key,
+                    parentValue: options.find(opt => opt.value === val),
+                    ignoreAliasError: false,
+                    ignoreSelectError: false,
+                } as IParentAlias);
+            });
+        }
+
+        this.setState(() => ({
+            parentOptions: options,
+            model: model.merge({parentAliases}) as SampleTypeModel
+        }));
+    };
+
+
+
+    //Generates a temporary id for add/delete of the import aliases
+    static generateAliasId() {
+        return generateId("sampleset-parent-import-alias-");
+    }
+
+    static getImportAliasesAsMap(model: SampleTypeModel): Map<string,string> {
```

Comment:
looks like this only uses the SampleTypeModel object as input, should this then be moved to a function on that model class?

Reply:
I had it there originally, but given its need to know the implementation details of this component I thought it was more appropriate here. Could be convinced otherwise.